### DISCUSSION TITLE
Force event report to SmartThings when GPIO14 state changes

### DISCUSSION THREAD

**Discussion by k0jdd** [arendst/Tasmota#17983] [Q&A] [answered]
I am sorry if the answer is in here somewhere. (I tried lots of keyword searches and could not find anything)

Situation: I am using a NodeMCU in a mailbox alert project. I am using GPIO14 as a dry contact. When the state of GPIO14 changes I want an event driven report sent to SmartThings. 

What happens is that the event does not get sent when the state changes -- I have to until the next update interval for ST to see the change in state.(up to a minute)  This is mostly an annoyance but I would prefer a more immediate notification.

Is there a way I can force the state change to be reported in a more timely manner? Is this a SmartThings issue or limitation?

**Current config**
NodeMCU ESP8266EX
Tasmota 12.3.1

![image](https://user-images.githubusercontent.com/46826320/219884139-707e07b8-10eb-4e3a-88f0-ab6437d71ab4.png)

**Console commands applied** 
SwitchMode1 1                (follow switch state)

I chose this config because even though I am not turning on/off any relays I can see the state in the WebUI.  I also tried using Generic0. That works the same as current config but I can't see the state in the WebUI.

**Observer behavior**
When I change the switch state  in the WebUI  the change shows up almost immediately in the ST App.
When I change the switch state  in the ST App  the change shows up almost immediately in the WebUI.
When i change the switch state using GPIO14 the change shows up almost immediately in the WebUI but it can take up to 30 seconds for it to show up in the ST App.  

I tried changing TelePeriod to no affect. That seems to only affect MQTT which I am not using.

I can live with it as-is on this project but It would be nice to know if there is a solution.

**Comment by sfromis** [comment]
You not seeing the state "timely" may be due to you not using a good technique to get it. But you told very little about how you implemented it. You seeing (and wanting) the state in the web UI is **not** the state of the switch, but the sate of the relay, where default actions of Tasmota couples the Switch input with the relay (similar to using the button on the Sonoff Basic). If you are ok with this, you "should" react to the `stat/tasmotatopic/POWER` messages, to get updates as things happens. Using MQTT is much recommended, but if not - you told nothing about how you get messages to SmartThings. Is SmartThings polling the device at intervals, or what are you doing? If you have a way of sending a message to SmartThings, you could do that with Tasmota rules.

Forwarding state can be done many ways, all depends on what limitations you have to make it harder, like if you decided against using MQTT, which is generally the easiest way.

**Reply by barbudor** [reply]
You said you are using a NodeMCU, so why are you configuring it as a Sonoff Basic as and you have no relay ?

You need to read the doc section about [Buttons -and-Switches](https://tasmota.github.io/docs/Buttons-and-Switches/) and how to capture switch events

You don't say how you are sending the event to SmartThings
Can SmartThing understand Tasmota MQTT messages ?

**Comment by k0jdd** [comment, marked as answer]
@barbudor  -  I tried several configurations including generic.  All yielded the same result.

@sfromis - Thanks for the feedback.  It helped me determine the underlying issues.


FWIW - Using _weblog 4_ I was able to determine the following:

No MQTT here. SmartThings is polling the device with a web command. That poll interval seems to be 45 seconds. You can force a poll from Startthings APP if you desire - but that doesn't help me.

I will pursue this further with the ST crew. 

Thanks.

UPDATE: 

I figured this out - Error code ID10T - ( Self inflicted )

This was all caused by doing things out of sequence. That’s why I call myself a NOOB.

**When you add the device to SmartThings using TASMOTA edge, it writes a massive rule to the device for reporting state changes to ST using WebQuery.**

If you hork around with the TASMOTA configuration after that, the rule can get clobbered any number of ways

So the correct sequence is.

    Get the device completely configured.
    Add the device to SmartThings
    LEAVE WELL ENOUGH ALONE.

I resolved the situation by deleting the device from ST and re-adding it.
